### ISSUE TITLE
APIObject doesn't respect context namespace

### ISSUE THREAD

**Issue by pehala** [openshift/openshift-client-python#112]
If you create `APIObject` with a model that is missing `namespace` it will default to the `default` namespace instead of whatever namespace context has.

**Comment by bradmwilliams** [comment]
I've tried to reproduce the issue, but the object is always created in the context that I am currently running under.
Could you share an example of what you are doing? 

Here is the script that Im running:
``` python
#!/usr/bin/env python

import openshift as oc
from openshift import APIObject


if __name__ == '__main__':
    print('Current Context: {}'.format(oc.get_config_context()))

    my_pod = {
        "apiVersion": "v1",
        "kind": "Pod",
        "metadata": {
            "name": "sample-pod",
            "labels": {}
        },
        "spec": {
            "containers": [
                {
                    "name": "sample-pod",
                    "image": "python:3",
                    "command": [
                        "tail",
                        "-f",
                        "/dev/null"
                    ]
                }
            ]
        }
    }

    pod_selector = oc.create(APIObject(my_pod))
    pod_selector.until_all(1, success_func=oc.status.is_pod_running)

    pods = oc.selector('pods').objects()
    print('Found {} pods'.format(len(pods)))
    print('Namespace: {}'.format(pods[0].namespace()))
```

**Comment by bradmwilliams** [comment]
So, if I select a context that does not have a `namespace` defined, then my pods go into the `default` project.  Other than that, objects are created in the namespace of the user's `current-context`. 

I'm happy to evaluate further, if necessary, but I'm of the opinion that this is working as expected.

**Comment by pehala** [comment]
The problem is not with the global context but it happens only when you override the context for APIObject like this:
```python
context = Context()
context.project_name = "my_own_project"

pod_selector = oc.create(APIObject(my_pod, context=context))
```
Here the context.project_name will be overridden and it will end up in the namespace from the global context.

Here is your example with that change
```python
#!/usr/bin/env python

import openshift as oc
from openshift import APIObject, Context

if __name__ == '__main__':
    print('Current Context: {}'.format(oc.get_config_context()))

    my_pod = {
        "apiVersion": "v1",
        "kind": "Pod",
        "metadata": {
            "name": "sample-pod",
            "labels": {}
        },
        "spec": {
            "containers": [
                {
                    "name": "sample-pod",
                    "image": "python:3",
                    "command": [
                        "tail",
                        "-f",
                        "/dev/null"
                    ]
                }
            ]
        }
    }

    context = Context()
    context.project_name = "my_own_project"

    pod_selector = oc.create(APIObject(my_pod, context=context))
    pod_selector.until_all(1, success_func=oc.status.is_pod_running)

    pods = oc.selector('pods').objects()
    print('Found {} pods'.format(len(pods)))
    print('Namespace: {}'.format(pods[0].namespace()))
```

**Comment by pehala** [comment]
Hey @bradmwilliams, thanks for the reply, I could definitely use context manager and it is how I fixed it, the use-case is factory method for creating instances of APIObject for custom CR. The reason I reported this is that I saw that APIObject has a context parameter and I wanted to take advantage of that.

I still believe this is a bug, however as you described a very minor one as this was never intended to be used like that, but since the fix is very simple I will submit PR to fix this.
